qiskit/aqua/operators/gradients/gradient_base.py

```diff
@@ -0,0 +1,68 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The base interface for Aqua's gradient."""
+
+from typing import Union
+
+from qiskit.aqua.operators.gradients.circuit_gradients import CircuitGradient
+from qiskit.aqua.operators.gradients.derivative_base import DerivativeBase
+
+
+class GradientBase(DerivativeBase):  # pylint: disable=abstract-method
+    """Base class for first-order operator gradient.
+
+    Convert an operator expression to the first-order gradient.
+    """
+
+    def __init__(self,
+                 grad_method: Union[str, CircuitGradient] = 'param_shift',
+                 **kwargs):
+        r"""
+        Args:
+            grad_method: The method used to compute the state/probability gradient. Can be either
+                ``'param_shift'`` or ``'lin_comb'`` or ``'fin_diff'``. Deprecated for observable
```

Comment:
What does 'Deprecated for observable gradient' mean? This is a first release of code so this seems odd. Is this related to the CircuitGradient that is not mentioned as one of the possible input values.

Reply:
Yes exactly.
So the observable gradient (w.r.t. observable parameters) is a special type of gradient that only requires the evaluation of a 'subpart' of the original observable but no circuit gradient (w.r.t. circuit parameters).
I will update the doctoring to clarify this.

Reply:
I guess the normal use we have of 'deprecated' is referring to some old way of doing things (some deprecated method for example)  where there is now a new way (some newer better method(s) for example). While the old way is not forbidden, its now not recommended. In this context I am trying to understand `Deprecated for gradients w.r.t observable parameters.` since if its deprecated (but its brand new function that has not even been released yet) what should I do instead. Maybe how I am interpreting it is not what you mean at all?